src/main/java/nextstep/subway/auth/ui/AuthenticationPrincipalArgumentResolver.java

```diff
@@ -28,6 +28,8 @@ public boolean supportsParameter(MethodParameter parameter) {
     public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
                                   NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
         String credentials = AuthorizationExtractor.extract(webRequest.getNativeRequest(HttpServletRequest.class));
-        return authService.findMemberByToken(credentials);
+        AuthenticationPrincipal annotation = parameter.getParameterAnnotation(AuthenticationPrincipal.class);
+
+        return authService.findMemberByToken(annotation, credentials);
```

Comment:
어노테이션을 인자로 사용하기보다는 required를 인자로 전달해주면 좋을 것 같아요 :)

```suggestion
        return authService.findMemberByToken(annotation.required(), credentials);
```

Reply:
이 부분 고민했었는데, 혹시 다른 정보다 더 들어올 일이 있을 때가 있지 않을까 하고 이렇게 넘겨 주었는데요~ 나중에 다시 추가할 일이 있을 때 변경해 주어도 될 것 같네요..!